File: aws/AWS_Lambda_in_Action/Chapter02/customGreetingsOnDemand.js
console.log('Loading function');

exports.handler = (event, context, callback) => {
    console.log('Received event:',
        JSON.stringify(event, null, 2));
    console.log('greet =', event.greet);
    console.log('name =', event.name);
    var greet = '';
    if ('greet' in event) {
        greet = event.greet;
    } else {
        greet = "Hello";
    }
    var name = '';
    if ('name' in event) {
        name = event.name;
    } else {
        name = "World";
    }
    var greetings = greet + ' ' + name + '!';
    console.log(greetings);
    callback(null, greetings);
};

File: aws/AWS_Lambda_in_Action/Chapter02/customGreetingsOnDemand.py
import json

print('Loading function')

def lambda_handler(event, context):
    print("Received event: " +
        json.dumps(event, indent=2))
    if 'greet' in event:
        greet = event['greet']
    else:
        greet = 'Hello'
    if 'name' in event:
        name = event['name']
    else:
        name = 'World'
    greetings = greet + ' ' + name + '!'
    print(greetings)
    return greetings
